[confluence] fix: On play/pause default control for play/pause

un1versal opened the pull request:
This is a bit odd, the default control https://github.com/xbmc/xbmc/blob/8abac52b32e220a12da0fd2dafd1fb6de8d24eb7/addons/skin.confluence/720p/VideoOSD.xml#L49 was removed with latest changes, re adding it back made no difference to behaviour.

But setting default control here restored it to how it use to work.

Essentially if you hit **_play/pause**_ on remote you dont want the default control to be **_chapter back  <id=100>**_ else if you press ok, instead of resuming you end up at beginning of file or previous chapter depending.

This restores this and therefore restores  WAF Factor on this, as I was getting annoyed that wife moaning this went back to beginning of file or previous chapter. Two weeks of listening to grief was my limit and therefore hence this PR.

@ronie

Commit:
[confluence] fix: On play/pause default control for play/pause 

This is a bit odd, the default control https://github.com/xbmc/xbmc/blob/8abac52b32e220a12da0fd2dafd1fb6de8d24eb7/addons/skin.confluence/720p/VideoOSD.xml#L49 was removed with latest changes, re adding it back made no difference to behaiviour.

But setting default control here restored it to how it use to work.

Essentially if you hit ***play/pause*** on remote you dont want the default control to be ***chapter back  <id=100>*** else if you press ok, instead of resuming you end up at beginning of file or previous chapter depending.

This restores this and therefore restores  WAF Factor on this, as I was getting annoyed that wife moaning this went back to beginning of file or previous chapter. Two weeks of listening to grief was my limit and therefore hence this PR.

@ronie

ronie commented:
that wouldn't work for LiveTV (when there's no control with id="202")

let's first get this one in https://github.com/xbmc/xbmc/pull/6303
and then think of a proper fix.

uNiversaI commented:
@ronie. Pretty sure I know what the problem is! The video osd use to use ***group control*** and now uses ***grouplist control***

***group control*** http://kodi.wiki/view/Group_Control supports which supports things like:

```<defaultcontrol always="true">202</defaultcontrol>```

The ***grouplist control*** http://kodi.wiki/view/Group_List_Control now in use doesn't seem to support 
this I have tried it. (hence I linked wiki)

If this is true also means that currently for ***live tv*** https://github.com/xbmc/xbmc/blob/master/addons/skin.confluence/720p/VideoOSD.xml#L191 wont work and if in that case 301 is desired it will go same route as this issue.

Im not sure if onup can be used for cheating around this or what but I thought Ide let you know what I found 

The best I could do is set the control I changed back to 100 and setting it to false. This if I understand the wiki right means the last used control is saved and will be the control selected next time you open osd.

For your consideration but seems if a proper solution is desired is to add <defaultcontrol> support to grouplist as well or some other solution, which I cant do anyway.

ronie commented:
a possible way to fix it:
https://github.com/xbmc/xbmc/pull/6319

Commit:
[confluence] osd save last used control & remove unsupported

Setting

<defaultcontrol always="false">100</defaultcontrol>

makes osd save the last used control and focus that next time you open
video OSD.

the issue is group list control doesnt support  defaultcontrol tag and
therefore this causes a problem when play/pause because no default
control can be set.

because of that this following is also removed.

<defaultcontrol always="true">301</defaultcontrol>

best solution is to use group control instead of group list or add
defaultcontrol support tag to group list.

ronie commented:
closing in favour of https://github.com/xbmc/xbmc/pull/6319
